### PR TITLE
Fixed the issue in role visibility

### PR DESCRIPTION
**Pull request by HiranyaKavishani**
## Purpose
This fixes for an issue in role visibility when multiple roles separate with spaces after the comma.

related Issues :
[#3980](https://github.com/wso2/product-apim/issues/3980)
[#5812](https://github.com/wso2/carbon-apimgt/issues/5812)
[#3858](https://github.com/wso2/product-apim/issues/3858)

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -2039,6 +2042,9 @@ public void addFileToDocumentation(APIIdentifier apiId, Documentation documentat
                 String[] visibleRoles = new String[0];
                 if (visibleRolesList != null) {
                     visibleRoles = visibleRolesList.split(",");
+                    for(int i = 0; i < visibleRoles.length; i++) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Formatting issues

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -1212,6 +1212,9 @@ private void updateApiArtifact(API api, boolean updateMetadata, boolean updatePe
 
                 if (visibleRolesList != null) {
                     visibleRoles = visibleRolesList.split(",");
+                    for(int i = 0; i < visibleRoles.length; i++) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Formatting issues

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 3]
```diff
@@ -2573,6 +2579,9 @@ public void updateDocumentation(APIIdentifier apiId, Documentation documentation
             String visibleRolesList = api.getVisibleRoles();
             if (visibleRolesList != null) {
                 authorizedRoles = visibleRolesList.split(",");
+                for(int i = 0; i < authorizedRoles.length; i++) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Formatting issues

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -2699,6 +2708,9 @@ protected void createAPI(API api) throws APIManagementException {
             String[] visibleRoles = new String[0];
             if (visibleRolesList != null) {
                 visibleRoles = visibleRolesList.split(",");
+                for(int i = 0; i < visibleRoles.length; i++) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Formatting issues

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java [hunk 5]
```diff
@@ -4063,6 +4063,9 @@ protected void checkAccessControlPermission(APIIdentifier identifier) throws API
             String storeVisibilityRoles = apiResource.getProperty(APIConstants.STORE_VIEW_ROLES);
             if (storeVisibilityRoles != null && !storeVisibilityRoles.trim().isEmpty()) {
                 String[] storeVisibilityRoleList = storeVisibilityRoles.split(",");
+                for(int i = 0; i < storeVisibilityRoleList.length; i++) {
```
Comment (mushthaq33) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Formatting issues

Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java]: Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 3 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 4 above)
Comment (uvindra) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Lets also use enhanced for loop(https://stackoverflow.com/a/18738765/3296350), its a lot cleaner

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 6]
```diff
@@ -2699,6 +2714,11 @@ protected void createAPI(API api) throws APIManagementException {
             String[] visibleRoles = new String[0];
             if (visibleRolesList != null) {
                 visibleRoles = visibleRolesList.split(",");
+                int index = 0;
+                for (String role: visibleRoles) {
+                    visibleRoles[index] = role.trim();
+                    index++;
+                }
```
Comment (TanyaM) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Same snippet repeated at many places. Can't we move this to a utility method and use elsewhere.

**Pull request merged**